Automattic/jetpack
# UI tests: Add MVP e2e configuration

This time for real. This PR adds a way to run ngrok powered WP installation in CI. Also includes a dummy test which tries to login into newly created WP site.  

I plan to have these tests disabled after this PR will be merged since it does not include any actual tests. In further PR's I'll plan to add some real tests

Few things that I not really sure about, and would like to get some feedback on these:
- ~travis config: stages vs one more parallel build. Former approach doubles build execution time (6min total). The overall e2e build time is 3 minutes, but it will grow as we'll add more tests. UPD: I updated travis config to run e2e job in parallel, and made it optional (e.g. failing job won't fail the build)~ Switched the job to run in parallel as the other jobs 
- ~dependencies in root `package.json`. E2E tests don't have much in common with root dependencies so it might be a good idea to move them into its own file. Not sure that we really need that.~ Decided not to complicate things for now.

#### Changes proposed in this Pull Request:

* add bash script to provision ngrok powered WP site in CI context
* add a dummy test to login into a newly created site  

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* p9dueE-Ju-p2

#### Testing instructions:






* All the magic should happen inside CI build. In a job log, you should see something similar to:
```
$ JEST_PUPPETEER_CONFIG=tests/e2e/jest-puppeteer.config.js jest --config tests/e2e/jest.config.js
 PASS  tests/e2e/specs/dummy.test.js (22.885s)
  First test
    ✓ Can login to wp-admin and click Connect on Jetpack page (22205ms)

  console.log tests/e2e/lib/pageHelper.js:10
    Found element by locator: #welcome-panel

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        22.943s
Ran all test suites matching /tests\/e2e\/specs\/dummy.test.js/i.
✨  Done in 24.25s.
```
* Try running test locally: `yarn test-e2e`. For this, you should have a local WP installation. The test expects that it's running on localhost, and your login creds are `wordpress`/`wordpress`. These can be tweaked by passing env vars along with the script eg:
`WP_BASE_URL="http://wpdev.localhost" WP_USERNAME=admin WP_PASSWORD=admin yarn test-e2e tests/e2e/specs/dummy.test.js`
* Check that `RUN_E2E` env variable toggles e2e test execution. Head to https://travis-ci.org/Automattic/jetpack/settings and add `RUN_E2E` variable set to `true`:
example build _without_ E2E: https://travis-ci.org/Automattic/jetpack/builds/533758743
example build _with_ E2E: https://travis-ci.org/Automattic/jetpack/builds/533754944

#### Proposed changelog entry for your changes:

* none
